smart_home/src/services/temperature_service.cc

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include "../../include/services/temperature_service.h"
+#include "../../include/sensors/temperature_sensor.h"
+#include "../../include/devices/ac_unit.h"
+
+namespace services {
+
+void TemperatureService::AddSensor(sensors::Sensor* sensor) {
+    sensors_.push_back(sensor);
+}
+
+void TemperatureService::AddDevice(devices::Device* device) {
+    devices_.push_back(device);
```

Comment:
same here..with a bit of checks, you could make sure all the devices are related to temperature service